src/main/java/com/google/api/codegen/transformer/java/JavaModelTypeNameConverter.java

```diff
@@ -273,14 +274,23 @@ private static String getResourceNamePackage(ResourceNameConfig resourceNameConf
       return commonResourceName;
     }
 
-    ResourceNameType resourceNameType = resourceNameConfig.getResourceNameType();
+    ResourceNameType resourceNameType = fieldConfig.getResourceNameConfig().getResourceNameType();
     switch (resourceNameType) {
       case ANY:
         return "com.google.api.resourcenames";
       case FIXED:
       case SINGLE:
       case ONEOF:
-        return getJavaPackage(resourceNameConfig.getAssignedProtoFile());
+        ProtoFile assignedProtoFile = fieldConfig.getResourceNameConfig().getAssignedProtoFile();
+        FieldModel fm = fieldConfig.getField();
+        if (fm instanceof ProtoField) {
+          ProtoFile fieldProtoFile = ((ProtoField) fm).getProtoField().getFile();
+          if (!assignedProtoFile.getFullName().equals(fieldProtoFile.getFullName())) {
+            assignedProtoFile = fieldProtoFile;
```

Comment:
This seems to be using the package with code that references a resource instead of defines a resource to generate Java package for the resource name classes, which is absolutely correct. However, two questions:

1) should this always be used?
2) I _think_ the way `assignedProtoFile` is set up makes sure it is always inside the package of the API we are generating GAPICs for. Is that not the case?

Reply:
1 & 2: If I understand your question correctly: I think `assignedProtoFile` cannot be used properly for our purpose (effectively that property should be removed, but I'm trying to keep this change as small as possible, since there are already to many "moving" parts everywhere and very easy to break already working things). Basically `assignedProtoFile` assigns annotation definition to the file it was defined in, as result it assumes `1:1` relationship between resource name definition and generated `<Resource>Name` classes.

Unfortunately the above statement is not true for java. If we have a resource name declared in one place, but used under two different packages (for example if API consists of 2 different interfaces under different packages) gapic-generator generates 2 `<Resource>Name`  classes from one resource name definition: `package1.<Resource>Name` and `package2.<Resource>Name`. This means that there is actual `1:m` relationship between resource name definition and the generated classes, but `assignedProtoFile` returns a single file per single resource (so the whole class model chosen here does not allow to express the relationship).

Reply:
Okay that makes sense. In this case why is it necessary to check whether assignedProtoFile.getFullName() and filedProtoFile.getProtoField().getFile() are equal? It seems that the latter can always be used if `fm instanceof ProtoField` is true.